src/solver/mha/mha_ck_fa_v2_solver_forward.cpp

```diff
@@ -0,0 +1,265 @@
+           && descsFwd.qDesc.GetType() == miopenHalf    //
+           && descsFwd.vDesc.GetType() == miopenHalf    //
+           && descsFwd.oDesc.GetType() == miopenHalf;   //
+#else
+    return false;
+#endif
+}
+
+std::size_t MhaCKFlashAttentionV2Forward::GetWorkspaceSize(
+    [[maybe_unused]] const ExecutionContext& context,
+    [[maybe_unused]] const miopen::mha::ProblemDescription& problem) const
+{
+    return 0;
+}
+
+ConvSolution
+MhaCKFlashAttentionV2Forward::GetSolution([[maybe_unused]] const ExecutionContext& context,
+                                          const miopen::mha::ProblemDescription& problem) const
+{
+#if MIOPEN_USE_COMPOSABLEKERNEL
+    auto result         = ConvSolution{miopenStatusSuccess};
+    result.workspace_sz = 0;
+
+    const miopen::mha::MhaInputDescsForward& descsFwd = problem.GetDescsForward();
+    auto [N_k, H_k, S_k, D_k] = miopen::tien<4>(descsFwd.kDesc.GetLengths());
+    auto [N_q, H_q, S_q, D_q] = miopen::tien<4>(descsFwd.qDesc.GetLengths());
+    auto [N_v, H_v, S_v, D_v] = miopen::tien<4>(descsFwd.vDesc.GetLengths());
+
+    ck_tile::index_t batch    = N_q;
+    ck_tile::index_t seqlen_q = S_q;
+    ck_tile::index_t seqlen_k = S_k;
+    ck_tile::index_t hdim_q   = D_q;
+    ck_tile::index_t hdim_v   = D_v;
+    ck_tile::index_t nhead    = H_q;
+    ck_tile::index_t nhead_k  = H_k;
+    ck_tile::index_t nhead_q  = H_q;
+
+    bool is_group_mode = false;
+    bool o_perm = true, i_perm = true; // if true, will be batch * nhead * seqlen * hdim
+
+    float scale_s = descsFwd.scale;
+    float scale_p = 1.0;
+    float scale_o = 1.0;
+
+    const ck_tile::index_t shape_seqlen_q = seqlen_q;
+    const ck_tile::index_t shape_seqlen_k = seqlen_k;
+    const ck_tile::index_t max_seqlen_q   = seqlen_q;
+    const ck_tile::index_t max_seqlen_k   = seqlen_k;
+
+    fmha_fwd_traits fmha_traits;
+    fmha_traits.hdim_q              = hdim_q;
+    fmha_traits.hdim_v              = hdim_v;
+    fmha_traits.data_type           = Convert(descsFwd.qDesc.GetType());
+    fmha_traits.is_group_mode       = is_group_mode;
+    fmha_traits.is_v_rowmajor       = false;
+    fmha_traits.mask_type           = mask_enum::no_mask;
+    fmha_traits.has_lse             = false;
+    fmha_traits.do_fp8_static_quant = false;
+    fmha_traits.has_dropout         = false;
+
+    fmha_fwd_args fmha_args;
+    fmha_args.batch          = batch;
+    fmha_args.hdim_q         = hdim_q;
+    fmha_args.hdim_v         = hdim_v;
+    fmha_args.nhead_q        = nhead_q;
+    fmha_args.nhead_k        = nhead_k;
+    fmha_args.stride_q       = (i_perm ? hdim_q : nhead * hdim_q);
+    fmha_args.stride_k       = (i_perm ? hdim_q : nhead_k * hdim_q);
+    fmha_args.stride_v       = (i_perm ? shape_seqlen_k : nhead_k * shape_seqlen_k);
+    fmha_args.batch_stride_q = (nhead * shape_seqlen_q * hdim_q);
+    fmha_args.batch_stride_k = (nhead_k * shape_seqlen_k * hdim_q);
+    fmha_args.batch_stride_v = (nhead_k * hdim_v * shape_seqlen_k);
```

Comment:
I think these stride values need to come from the tensorDescriptors, and not calculated actually.